src/main/java/com/bootcamp/demo/data_access/AccountRepository.java

```diff
@@ -0,0 +1,21 @@
+package com.bootcamp.demo.data_access;
+
+import com.bootcamp.demo.model.Account;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firebase.cloud.FirestoreClient;
+import org.springframework.stereotype.Repository;
+
+import java.util.concurrent.ExecutionException;
+
+@Repository
+public class AccountRepository implements AbstractRepository<Account> {
+    public String add(final Account account) throws ExecutionException, InterruptedException {
+        Firestore db = FirestoreClient.getFirestore();
+        DocumentReference docRef = db.collection("accounts").document(account.getUsername());
```

Comment:
What happens if we are trying to add an account with the same username but with different attributes?